### ISSUE TITLE
Consider reverting `cursor::MoveToColumn(0)` non-zero handling or changing it?

### ISSUE THREAD

**Issue by dsherret** [crossterm-rs/crossterm#683]
**Is your feature request related to a problem? Please describe.**

I was using `MoveToColumn(0)` because I assumed it was 0-indexed, but in https://github.com/crossterm-rs/crossterm/pull/627 it was changed to be silently ignored so the output of my code started acting funny when I upgraded crossterm.

It seems better to just let `0` be the same as `1` for `MoveToColumn`? I think some other people might also get confused why it's not working as well and blame crossterm on being buggy.

**Describe the solution you'd like**

`MoveToColumn(0)` is equivalent to `1` or perhaps panics? Alternatively, maybe this should only accept a NonZeroU16, though that api might not be so nice. `0` equivalent to `1` seems most ideal.

I can understand how this current functionality is desired for what's described in #626, but not so much for this.

**Comment by Johann150** [comment]
I think it would make more sense if they were zero indexed because `MoveTo` and `cursor::position` are zero indexed, and this is documented. The phrasing of the documentation lead me to believe that the statement was true for the whole API, which is confusing:
> Top left cell is represented as `0,0`.
-- https://docs.rs/crossterm/0.23.2/crossterm/cursor/struct.MoveTo.html#notes

**Comment by TimonPost** [comment]
I see there is correct confusing how the API operates and in being consistent. The aim is to be 0-based but ANSI codes are often 1-based. `MoveTo` internally increments the passed parameters to conform that. 

![image](https://user-images.githubusercontent.com/19969910/176745085-7888c444-231c-412f-be32-0cf7576e0e73.png)

I think this is indeed something to be reverted. We should then just increment the parameter by one and accept 0 parametes everywere, in all cursor functions.

**Comment by TimonPost** [comment]
Previous time I decided that it was confusing having Move Left by 0 makes the terminal move left by 1. But if you see it like: Move left to the zero'th cell then it makes more sense.

**Comment by TimonPost** [comment]
So the question is more about do we want all cursor functions to be consistent

**Comment by Johann150** [comment]
For me it does not make sense to adjust relative moves in this way. Something like "go two steps left" always has the same meaning, no matter if the coordinate system starts at `0,0` or `1,1`. I would be very confused if I say "go two steps left" and the cursor goes *three* steps left.

Either way, documenting it properly would be good.

**Comment by TimonPost** [comment]
Yes that was my previous thought as well. Lets do it like that, thus 1 is really moving 1. And document that this is the case rather than being consistent with moveto commands that people were confused by.
